pkgs/by-name/pr/protonmail-bridge-gui/package.nix

```
@@ -0,0 +1,134 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, pkg-config
+, libsecret
+, cmake
+, ninja
+, qt6
+, grpc
+, protobuf
+, zlib
+, gtest
+, sentry-native
+, protonmail-bridge
+}:
+
+let
+  pname = "protonmail-bridge-gui";
+  inherit (protonmail-bridge) version src;
+
+  patches = [
+    ./use-nix-googletest.patch
+    ./designator-order-fix.patch
+  ];
+
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DBRIDGE_APP_FULL_NAME='Proton Mail Bridge'"
+    "-DBRIDGE_VENDOR='Proton AG'"
+    "-DBRIDGE_REVISION=${src.rev}"
+    "-DBRIDGE_TAG=v${version}"
+    "-DBRIDGE_BUILD_ENV=dev"
+    "-DBRIDGE_APP_VERSION=${version}"
+  ];
+
+in
+stdenv.mkDerivation {
+  inherit pname version src patches cmakeFlags;
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    ninja
+    qt6.qtbase
+    qt6.qtdeclarative
+    qt6.qtwayland
+    qt6.qtsvg
+    qt6.wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    libsecret
+    grpc
+    protobuf
+    zlib
+    gtest
+    sentry-native
+  ];
+
+  postPatch = ''
+    # bypass `vcpkg` by deleting lines that `include` internal/frontend/bridge-gui/BridgeSetup.cmake
+    find internal/frontend/bridge-gui -type f -name "CMakeLists.txt" -exec sed -i "/BridgeSetup\\.cmake/d" {} \;
+
+    # modify DeployLinux.cmake to use the available ICU version
+    sed -i 's/libicu\(i18n\|uc\|data\)\.so\.56/libicu\1.so/g' internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake
+  '';
+
+  dontUseCmakeConfigure = true; # disable the automatic configure phase
+
+  configurePhase = ''
+    runHook preConfigure
+
+    export QT_PLUGIN_PATH="${qt6.qtbase}/${qt6.qtbase.qtPluginPrefix}"
+
+    pushd internal/frontend/bridge-gui
+    cmake \
+      ${lib.concatStringsSep " " cmakeFlags} \
+      -G Ninja \
+      -S . \
+      -B build
+    popd
+
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    pushd internal/frontend/bridge-gui
+    cmake --build build --verbose
+    popd
+
+    runHook postBuild
+  '';
```

```suggestion
  sourceRoot = "source/internal/frontend/bridge-gui";

  env.QT_PLUGIN_PATH="${qt6.qtbase}/${qt6.qtbase.qtPluginPrefix}";

  postPatch = ''
    # bypass `vcpkg` by deleting lines that `include` BridgeSetup.cmake
    find . -type f -name "CMakeLists.txt" -exec sed -i "/BridgeSetup\\.cmake/d" {} \;

    # modify DeployLinux.cmake to use the available ICU version
    sed -i 's/libicu\(i18n\|uc\|data\)\.so\.56/libicu\1.so/g' bridge-gui/DeployLinux.cmake
  '';
```

Setting `sourceRoot` directly avoids the boilerplate.
Not sure why `QT_PLUGIN_PATH` needs to be specified?

Naturally `installPhase` and the patches will also have to be updated accordingly, as relative dirs will be changed.

---

I resorted to all this ceremony for finer control, I should have refactored like this after getting it to build. See my latest force pushed commit, it still works!